### PR TITLE
udev: drop unused udev structs and use verbs in udevadm

### PR DESCRIPTION
**Pull request by yuwata**
This is a continuation of #9920, and part of #9731.

### CODE REVIEW

**Comment by evverx** [comment]
This pull request **introduces 1 alert** when merging 9ad7bb5d06ded15af79348417350e69edc9cb494 into e6a2feb57d60a8f8f5f64fdc6d9a23260fe0df05 - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-d97df60a17c4e2a9c6597a1525894cc11e7833ea)

**new alerts:**

* 1 for Missing header guard

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/udev/udev-builtin-kmod.c [hunk 1]
```diff
@@ -48,19 +48,19 @@ static int builtin_kmod_init(struct udev *udev) {
                 return -ENOMEM;
 
         log_debug("Load module index");
-        kmod_set_log_fn(ctx, udev_kmod_log, udev);
+        kmod_set_log_fn(ctx, udev_kmod_log, NULL);
```
Comment (keszybz) [inline src/udev/udev-builtin-kmod.c:51]: `udev_kmod_log` will get NULL as the first argument, which it ignores, so this is OK.

File: src/udev/udevadm.c [hunk 2]
```diff
@@ -95,34 +65,66 @@ int main(int argc, char *argv[]) {
...
+                { "info",         VERB_ANY, VERB_ANY, 0, info_main    },
+                { "trigger",      VERB_ANY, VERB_ANY, 0, trigger_main },
+                { "settle",       VERB_ANY, VERB_ANY, 0, settle_main  },
+                { "control",      VERB_ANY, VERB_ANY, 0, control_main },
+                { "monitor",      VERB_ANY, VERB_ANY, 0, monitor_main },
+                { "hwdb",         VERB_ANY, VERB_ANY, 0, hwdb_main    },
+                { "test",         VERB_ANY, VERB_ANY, 0, test_main    },
+                { "test-builtin", VERB_ANY, VERB_ANY, 0, builtin_main },
+                { "version",      VERB_ANY, VERB_ANY, 0, version_main },
+                { "help",         VERB_ANY, VERB_ANY, 0, help_main    },
```
Comment (keszybz) [inline src/udev/udevadm.c:102]: All the verb functions do argument parsing on their own, so using `VERB_ANY`, `VERB_ANY` seems correct.

File: src/udev/udevadm-hwdb.c [hunk 3]
```diff
@@ -569,139 +671,59 @@ int hwdb_main(int argc, char *argv[], void *userdata) {
...
-out:
-        if (trie) {
-                if (trie->root)
-                        trie_node_cleanup(trie->root);
-                if (trie->strings)
-                        strbuf_cleanup(trie->strings);
-                free(trie);
-        }
-        return rc;
+        return 0;
```
Comment (keszybz) [inline src/udev/udevadm-hwdb.c:728]: Hmm, something seems to be wrong here
```console
$ build/udevadm hwdb --update
Failed to write database : Invalid argument
free(): invalid pointer
[3]    25648 abort (core dumped)  build/udevadm hwdb --update
```

File: src/udev/udevadm-trigger.c [hunk 4]
```diff
@@ -323,7 +292,7 @@ int trigger_main(int argc, char *argv[], void *userdata) {
         }
         r = exec_list(udev_enumerate, action, settle_set);
         if (r < 0)
-                return 1;
+                return r;
```
Comment (keszybz) [inline src/udev/udevadm-trigger.c:295]: This changes the return convention... That's a break in compat, but hopefully nobody uses this return value. At least it's not documented.

File: src/udev/udevadm-control.c [hunk 5]
```diff
@@ -62,108 +64,103 @@ int control_main(int argc, char *argv[], void *userdata) {
...
 
                         if (DIV_ROUND_UP(s, USEC_PER_SEC) > INT_MAX)
-                                log_error("Timeout value is out of range.");
+                                log_error("Timeout value is out of range, ignoring.");
                         else {
-                                seconds = s != USEC_INFINITY ? (int) DIV_ROUND_UP(s, USEC_PER_SEC) : INT_MAX;
-                                timeout = seconds;
-                                rc = 0;
+                                timeout = s != USEC_INFINITY ? (int) DIV_ROUND_UP(s, USEC_PER_SEC) : INT_MAX;
                         }
```
Comment (keszybz) [inline src/udev/udevadm-control.c]: You can drop the braces here...

**Comment by keszybz** [comment]
Looks good, but I found one crash.

**Comment by yuwata** [comment]
Updated.
- double free and invalid path setting in `udevadm hwdb --update` are fixed.
- one coding style issue is fixed.

**Comment by keszybz** [comment]
I won't re-review, `-ETOOMANYPATCHESINSERIES` ;) I'm just checking the changed parts, and it looks all good.

**Pull request merged**